src/hero_bringup/toyota_software_utils.py

```diff
@@ -0,0 +1,108 @@
+import os
+import rospkg.common
+import rospkg.rospack
+from xml.etree.ElementTree import ElementTree
+import yaml
+
+global RP
+RP = None
+
+
+def read_dict_yaml(file_path, log):
+    # type: (str, callable) -> dict
+    """
+    Read a yaml file, which should be a dict at toplevel
+
+    :param file_path: yaml file to read
+    :type file_path: str
+    :param log: Log function to use, must accept a string
+    :type log: callable
```

Comment:
Why pass a log function along? In case of standard Python logging, this shouldn't be necessary. On the other hand, why not use `rospy`?

Reply:
I don't feel to use rospy for logging in a script which doesn't use any ros connection.

Reply:
So why not use 'standard' Python logging? In that case you could add a 'ROS' handler if you want to use `rospy` and a streamhandler if you don't

Reply:
I am not familiar with python logging but I will look into it